fem/fe.cpp

```diff
@@ -6598,6 +6598,37 @@ void Poly_1D::CalcChebyshev(const int p, const double x, double *u, double *d)
    }
 }
 
+void Poly_1D::CalcChebyshev(const int p, const double x, double *u, double *d,
+                            double *dd)
+{
+   // recursive definition, z in [-1,1]
+   // T_0(z) = 1,  T_1(z) = z
+   // T_{n+1}(z) = 2*z*T_n(z) - T_{n-1}(z)
+   // T'_n(z) = n*U_{n-1}(z)
+   // U_0(z) = 1  U_1(z) = 2*z
+   // U_{n+1}(z) = 2*z*U_n(z) - U_{n-1}(z)
+   // U_n(z) = z*U_{n-1}(z) + T_n(z) = z*T'_n(z)/n + T_n(z)
+   // T'_{n+1}(z) = (n + 1)*(z*T'_n(z)/n + T_n(z))
```

Comment:
Differentiating this w.r.t. `z` we can get the formula:
```
T''_{n+1}(z) = (n + 1)*((T'_n(z) + z*T''_n(z))/n + T'_n(z)) = (n + 1)*((n + 1)*T'_n(z) + z*T''_n(z))/n
```
which seems simpler than the one used below. Also, it has no issue at `z = ±1` which makes it a better choice for me. Is there some reason to prefer the expression below?

Reply:
The expression below I derived from differentiating the trigonometric form of the Chebyshev polynomials, thus the issue at `z=+-1`. In order to obtain the same values for your expression as for the one below, I had to add a small modification:
`
T''_{n+1}(z) = (n + 1)*(2*(n + 1)*T'_n(z) + z*T''_n(z)) / n
`

Reply:
Yes, the change in the formula is necessary to account for the rescaling `z = 2*x - 1`.

Thanks @aschaf.